tools/number-converter/src/NumberConverter.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+	<div class="NumberConverter">
+
+		<tool-markdown :markdown="`
+Enter a number to convert:
+		`"/>
+
+		<tool-error
+			v-if="errorMessage"
+			class="NumberConverter__errorMessage"
+			:message="errorMessage"/>
+
+		<tool-code
+			class="NumberConverter__input"
+			v-model="num"
+			:options="codeOptions"
+			:autoSize="true"/>
+
+		<tool-taskbar v-if="num">
+
+			<template v-if="!converted">
+				<tool-button
+					size="lg"
+					class="NumberConverter__button"
+					v-text="'Convert'"
+					@click.native="convert"/>
+				<b-form-radio-group
+				class="NumberConverter__button"
+				v-model="base"
+				:options="baseOptions"
+				buttons
+				button-variant="outline-secondary"
+				size="sm"/>
+			</template>
+
+			<template v-else>
+				<tool-button
+					size="lg"
+					class="NumberConverter__button"
+					v-text="'Copy Output'"
+					v-clipboard="num"/>
+				<tool-button
+					size="sm"
+					class="NumberConverter__button"
+					variant="secondary"
+					v-text="'Reset'"
+					@click.native="reset"/>
+			</template>
+		</tool-taskbar>
+
+	</div>
+</template>
+
+<script>
+export default {
+	data() {
+		return {
+			converted: false,
+			errorMessage: null,
+			num: '',
+			base: 10,
+			baseOptions: [
+				{ text: 'Binary', value: 2 },
+				{ text: 'Octal', value: 8 },
+				{ text: 'Hexadecimal', value: 16 }
+			]
+		}
+	},
+	computed: {
+		codeOptions() {
+			return {
+				readOnly: !!this.formatted,
+				lineNumbers: false,
+			};
+		}
+	},
+	methods: {
+		convert() {
+			try {
+				this.errorMessage = false;
```

Comment:
For readability + extensibility do you think it's worth seperating the conversion logic into different methods for each type of conversion? i.e. `convertToHex(...)` or something like that

Reply:
We could do but might have to change how the function is set up

Reply:
I think it's better to refactor this function now rather than in 6 months when it's harder